src/System/refine.rs

```diff
@@ -330,7 +331,9 @@ fn build_state_pair<'a>(
     ) {
         new_sp_zone = test_zone2;
     }
+
     new_sp.zone = new_sp_zone;
+    new_sp.calculate_max_bound(sys1, sys2, is_state1);
```

Comment:
Do we need to calculate max bounds every time we make a state pair? Would be nice if we could somehow save it on the SystemRepresentations instead

Reply:
I have left this for when we tackle issue #36